### PR TITLE
Etc Cleanup and more tests.

### PR DESCRIPTION
**Pull request by jacobb**

### CODE REVIEW

File: nap/engine.py [hunk 1]
```diff
@@ -244,8 +244,11 @@ def filter(self, **lookup_vars):
             obj_list = r_data
             extra_data = {}
 
-        if not hasattr(obj_list, '__iter__'):
-            raise ValueError('expected array-type response')
+        if obj_list:
+            try:
+                getattr(obj_list[0], 'keys')
```
Comment (gaconnet) [inline nap/engine.py]: Since 'keys' is fixed and not variable, why not `obj_list[0].keys`?

**Comment by gaconnet** [comment]
Nice big bunch of tests. Looks good!

File: docs/_themes/LICENSE [hunk 2]
```diff
@@ -0,0 +1,37 @@
+Copyright (c) 2010 by Armin Ronacher.
```
Comment (neruson) [inline docs/_themes/LICENSE:1]: lol

Comment (jacobb) [inline docs/_themes/LICENSE:1]: It’s the flask theme (which he BSD’d), so I keep the license.  

Best,
Jacob

Comment (neruson) [inline docs/_themes/LICENSE:1]: This is why you read to the bottom before commenting. :embarrassed:

**Comment by neruson** [comment]
Looks good to me!

**Comment by ethurgood** [comment]
Looks good

File: nap/exceptions.py [hunk 3]
```diff
@@ -1,8 +1,13 @@
+
```
Comment (fcurella) [inline nap/exceptions.py:1]: Nitpicking on PEP8: why the 2 blank lines at the beginning of the file?

**Comment by fcurella** [comment]
LGTM :cake: :sparkles:

**Pull request merged**